File: apps/server/src/api/components/user/user-models.ts
type UserRequest = {
  name: string
  email: string
}

type User = {
  id?: string | undefined
  name: string
  email: string
}

type UserList = User[]

type UserDTO = {
  id: string
  name: string
  email: string
}

type UserListDTO = UserDTO[]

export { User, UserDTO, UserList, UserListDTO, UserRequest }

File: apps/server2/src/api/components/user/user_models.py
import datetime

from pydantic import BaseModel


class UserRequest(BaseModel):
    name: str
    email: str


class User(BaseModel):
    id: str | None = None
    name: str
    email: str
    created_at: datetime.datetime | None = None
    updated_at: datetime.datetime | None = None


class UserResponse(BaseModel):
    id: str
    name: str
    email: str
    created_at: datetime.datetime
    updated_at: datetime.datetime | None = None
